internal/pkg/partition/wipe.go

```diff
@@ -76,14 +82,54 @@ func (v *VolumeWipeTarget) Wipe(ctx context.Context, log func(string, ...any)) e
 
 	defer parentBd.Unlock() //nolint:errcheck
 
+	if err := v.wipeParentLocked(log); err != nil {
+		return fmt.Errorf("error wiping device %q: %w", v.devName, err)
+	}
+
+	if parentBd == nil || v.partitionIndex == 0 {
+		return fmt.Errorf("missing parent block device or partition index")
+	}
+
+	if err := v.dropParentLocked(parentBd, log); err != nil {
+		return fmt.Errorf("error dropping partition: %w", err)
+	}
+
+	return nil
+}
+
+func (v *VolumeWipeTarget) wipeParentLocked(log func(string, ...any)) error {
```

Comment:
nit, should it be better with `wipeWithParentLocked` (and same for drop), as I read it first as "wipe parent"?